ImperatorToCK3Tests/MapperTests/RegionMapper/CK3RegionMapperTests.cpp

```diff
@@ -0,0 +1,232 @@
+	landedTitles.loadTitles(landedTitlesStream);
+	std::stringstream regionStream;
+	regionStream << "test_region = { duchies = { d_aquitane } }";
+	std::stringstream islandRegionStream;
+	try
+	{
+		theMapper.loadRegions(landedTitles, regionStream, islandRegionStream);
+		FAIL();
+	}
+	catch (const std::runtime_error& e)
+	{
+		ASSERT_STREQ("Region's test_region duchy d_aquitane does not exist!", e.what());
+	}
+}
+TEST(Mappers_RegionMapperTests, loadingBrokenCountyWillThrowException)
+{
+	mappers::CK3RegionMapper theMapper;
+	CK3::LandedTitles landedTitles;
+	std::stringstream landedTitlesStream;
+	landedTitlesStream << "k_anglia = { d_aquitane = { c_mers_broken = { b_hgy = { province = 69 } } } } \n";
+	landedTitles.loadTitles(landedTitlesStream);
+	std::stringstream regionStream;
+	regionStream << "test_region = { counties = { c_mers } }";
+	std::stringstream islandRegionStream;
+	try
+	{
+		theMapper.loadRegions(landedTitles, regionStream, islandRegionStream);
+		FAIL();
+	}
+	catch (const std::runtime_error& e)
+	{
+		ASSERT_STREQ("Region's test_region county c_mers does not exist!", e.what());
+	}
+}
+
+TEST(Mappers_RegionMapperTests, locationServicesWork)
+{
+	mappers::CK3RegionMapper theMapper;
+	CK3::LandedTitles landedTitles;
+	std::stringstream landedTitlesStream;
+	landedTitlesStream << "d_aquitane = { c_mers = { b_hgy = { province = 69 } } }";
+	landedTitles.loadTitles(landedTitlesStream);
+	std::stringstream regionStream;
+	regionStream << "test_region = { duchies = { d_aquitane } }\n";
+	regionStream << "test_region_bigger = { regions = { test_region } }\n";
+	regionStream << "test_region_biggest = { regions = { test_region_bigger } }\n";
+	std::stringstream islandRegionStream;
+	theMapper.loadRegions(landedTitles, regionStream, islandRegionStream);
+
+	ASSERT_TRUE(theMapper.provinceIsInRegion(69, "c_mers"));
+	ASSERT_TRUE(theMapper.provinceIsInRegion(69, "d_aquitane"));
+	ASSERT_TRUE(theMapper.provinceIsInRegion(69, "test_region"));
+	ASSERT_TRUE(theMapper.provinceIsInRegion(69, "test_region_bigger"));
+	ASSERT_TRUE(theMapper.provinceIsInRegion(69, "test_region_biggest"));
+}
+
+TEST(Mappers_RegionMapperTests, locationServicesCorrectlyFail)
+{
+	mappers::CK3RegionMapper theMapper;
+	CK3::LandedTitles landedTitles;
+	std::stringstream landedTitlesStream;
+	landedTitlesStream << "d_testduchy = { 1 2 3 } \n";
+	landedTitlesStream << "d_testduchy2 = { 4 5 6 } ";
+	landedTitles.loadTitles(landedTitlesStream);
+	std::stringstream regionStream;
+	regionStream << "test_region = { duchies = { d_testduchy } }";
+	regionStream << "test_region2 = { duchies = { d_testduchy2 } }\n";
+	std::stringstream islandRegionStream;
+	theMapper.loadRegions(landedTitles, regionStream, islandRegionStream);
+
+	ASSERT_FALSE(theMapper.provinceIsInRegion(4, "d_testduchy"));
+	ASSERT_FALSE(theMapper.provinceIsInRegion(5, "test_region"));
+	ASSERT_FALSE(theMapper.provinceIsInRegion(6, "test_superregion"));
```

Comment:
you don't have superregion defined so this test falls under nonsense test below.